### PR TITLE
[AND-3670] Update to 6.10.3 and add support for COPPA API

### PR DESCRIPTION
**Pull request by jiancchen**
Admob passes us a RequestConfiguration that shows what the pub selects for COPPA:
https://developers.google.com/admob/android/targeting#child-directed_setting

We use that to set the COPPA status

### CODE REVIEW

File: ThirdPartyAdapters/vungle/vungle/src/main/java/com/google/ads/mediation/vungle/VungleInitializer.java [hunk 1]
```diff
@@ -109,6 +110,17 @@ public void onAutoCacheAdAvailable(String placementId) {
...
   }
 
+  public void updateCoppaStatus(int configuration){
+    switch (configuration) {
+      case RequestConfiguration.TAG_FOR_CHILD_DIRECTED_TREATMENT_TRUE :
+        Vungle.updateUserCoppaStatus(true);
+        break;
+      case RequestConfiguration.TAG_FOR_CHILD_DIRECTED_TREATMENT_FALSE :
+        Vungle.updateUserCoppaStatus(false);
+        break;
```
Comment (HuFangshuai) [inline ThirdPartyAdapters/vungle/vungle/src/main/java/com/google/ads/mediation/vungle/VungleInitializer.java:120]: How about the default value `TAG_FOR_CHILD_DIRECTED_TREATMENT_UNSPECIFIED`?

**Review by HuFangshuai — COMMENTED** [review]
@jiancchen @palrahul Just a reminder: the 6.10.x-release target branch included Open Bidding feature which is in testing and not published. So if you want to publish COPPA changes seprately you can change the target branch to master.

File: ThirdPartyAdapters/vungle/vungle/src/main/java/com/google/ads/mediation/vungle/VungleInitializer.java (hunk 1 above)
Comment (jiancchen) [reply ThirdPartyAdapters/vungle/vungle/src/main/java/com/google/ads/mediation/vungle/VungleInitializer.java:120]: @HuFangshuai For that value, there's no specified behavior which mirrors our unset state. "TAG_FOR_CHILD_DIRECTED_TREATMENT_UNSPECIFIED  if you don't want to indicate how you would like your content treated with respect to COPPA in ad requests." In this case we won't send a COPPA value in our requests.

There is an issue though. If a developer sets true/false, there's no way for them to set TAG_FOR_CHILD_DIRECTED_TREATMENT_UNSPECIFIED afterwards.

**Comment by jiancchen** [comment]
@HuFangshuai gotcha, will rebase this against master

File: ThirdPartyAdapters/vungle/vungle/src/main/java/com/google/ads/mediation/vungle/VungleInitializer.java [hunk 2]
```diff
@@ -109,6 +110,17 @@ public void onAutoCacheAdAvailable(String placementId) {
     // Unused
   }
 
+  public void updateCoppaStatus(int configuration){
```
Comment (anton-blashchuk-zoolatech) [inline ThirdPartyAdapters/vungle/vungle/src/main/java/com/google/ads/mediation/vungle/VungleInitializer.java:113]: minor: not formatted

File: ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java [hunk 3]
```diff
@@ -65,6 +65,9 @@ public void requestInterstitialAd(@NonNull Context context,
       return;
     }
 
+    VungleInitializer.getInstance()
+            .updateCoppaStatus(mediationAdRequest.taggedForChildDirectedTreatment());
+
```
Comment (anton-blashchuk-zoolatech) [inline ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java:70]: what about com.google.ads.mediation.vungle.rtb.VungleRtbInterstitialAd ? don't we need same lines there? and in com.google.ads.mediation.vungle.rtb.VungleRtbBannerAd too

**Review by palrahul — COMMENTED** [review]
@jiancchen its good that we have a COPPA related settings that we can tie to pub making request for any AdUnit , we should use it, seems its tricking down to adapter only on specific AdRequest. Should we also consider adding option for Pub to explicitly set the COPPA flag directly on SDK , as seems from this solution until Pub makes any explicit request, all the remote APIs such as config, and any auto-cached Ad request would have already happened and that flag set by Pub would not be effective for those remote API requests

**Comment by jiancchen** [comment]
> @jiancchen its good that we have a COPPA related settings that we can tie to pub making request for any AdUnit , we should use it, seems its tricking down to adapter only on specific AdRequest. Should we also consider adding option for Pub to explicitly set the COPPA flag directly on SDK , as seems from this solution until Pub makes any explicit request, all the remote APIs such as config, and any auto-cached Ad request would have already happened and that flag set by Pub would not be effective for those remote API requests

@palrahul I think that's a good idea, we can add an API to VungleNetworkSettings since our initial config request may not have the coppa status

**Pull request closed without merging**